clients/urlclient/interfaces/retry.go

```diff
@@ -0,0 +1,32 @@
+/*******************************************************************************
+ * Copyright 2020 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+// interfaces defines contracts specific to the urlclient package
+package interfaces
+
+// RetryStrategy defines some way to verify that a RegistryClient has received data.
+// It accomplishes this by monitoring the values pointed to in its parameters,
+// which communicate the state of the RegistryClient.
+// When the values pointed to by the parameters are in the desired state,
+// the URL that should be passed back to the RegistryClient should be returned by the RetryStrategy.
+type RetryStrategy interface {
```

Comment:
I do not know if my previous comments  still apply: https://github.com/edgexfoundry/go-mod-core-contracts/pull/207#discussion_r373183961

Reply:
I think it's a neat idea, but it puts so much onus on the consumer to generate implementation that I don't want to do it. I've been playing with this idea today and I couldn't find an iteration I was happy with that balanced leaking implementation (what we're doing, tbh, with the lock state stuff that exists to aid testing) and something like your proposal that swings back to the other extreme.

Reply:
Agree this could be better structured.  Going to dismiss your review so we can get this code merged.